aws/resource_aws_wafregional_web_acl_association_test.go

```diff
@@ -0,0 +1,161 @@
+	var webAcl waf.WebACL
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckWafRegionalWebAclAssociationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckWafRegionalWebAclAssociationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafRegionalWebAclAssociationExists("aws_wafregional_web_acl_association.foo", &webAcl),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckWafRegionalWebAclAssociationDestroy(s *terraform.State) error {
+	return testAccCheckWafRegionalWebAclAssociationDestroyWithProvider(s, testAccProvider)
+}
+
+func testAccCheckWafRegionalWebAclAssociationDestroyWithProvider(s *terraform.State, provider *schema.Provider) error {
+	conn := provider.Meta().(*AWSClient).wafregionalconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_wafregional_web_acl_association" {
+			continue
+		}
+
+		web_acl_id, resource_arn := resourceAwsWafRegionalWebAclAssociationParseId(rs.Primary.ID)
+
+		resp, err := conn.ListResourcesForWebACL(&wafregional.ListResourcesForWebACLInput{WebACLId: aws.String(web_acl_id)})
+		if err != nil {
+			found := false
+			for _, list_resource_arn := range resp.ResourceArns {
+				if resource_arn == *list_resource_arn {
+					found = true
+					break
+				}
+			}
+			if found {
+				return fmt.Errorf("WebACL: %v is still associated to resource: %v", web_acl_id, resource_arn)
+			}
+		}
+	}
+	return nil
+}
+
+func testAccCheckWafRegionalWebAclAssociationExists(n string, webAcl *waf.WebACL) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		return testAccCheckWafRegionalWebAclAssociationExistsWithProvider(s, n, webAcl, testAccProvider)
+	}
+}
+
+func testAccCheckWafRegionalWebAclAssociationExistsWithProvider(s *terraform.State, n string, webAcl *waf.WebACL, provider *schema.Provider) error {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return fmt.Errorf("Not found: %s", n)
+	}
+
+	if rs.Primary.ID == "" {
+		return fmt.Errorf("No WebACL association ID is set")
+	}
+
+	web_acl_id, resource_arn := resourceAwsWafRegionalWebAclAssociationParseId(rs.Primary.ID)
+
+	conn := provider.Meta().(*AWSClient).wafregionalconn
+	resp, err := conn.ListResourcesForWebACL(&wafregional.ListResourcesForWebACLInput{WebACLId: aws.String(web_acl_id)})
+	if err != nil {
+		return fmt.Errorf("List Web ACL err: %v", err)
+	}
+
+	found := false
+	for _, list_resource_arn := range resp.ResourceArns {
+		if resource_arn == *list_resource_arn {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("Web ACL association not found")
+	}
+
+	return nil
+}
+
+const testAccCheckWafRegionalWebAclAssociationConfig = `
+resource "aws_wafregional_ipset" "foo" {
+  name = "foo"
+  ip_set_descriptors {
+    type = "IPV4"
+    value = "192.0.7.0/24"
+  }
+}
+
+resource "aws_wafregional_rule" "foo" {
+  depends_on = ["aws_wafregional_ipset.foo"]
```

Comment:
The `depends_on` is redundant here as the relationship is already defined by `data_id = "${aws_wafregional_ipset.foo.id}"` below.